include/deal.II/lac/read_write_vector.templates.h

```diff
@@ -42,13 +46,16 @@ namespace LinearAlgebra
         if (val != NULL)
           free(val);
         val = NULL;
+        thread_loop_partitioner.reset(new parallel::internal::TBBPartitioner());
       }
     else
       {
         if (val != NULL)
           free(val);
 
         Utilities::System::posix_memalign ((void **)&val, 64, sizeof(Number)*new_alloc_size);
+        if (new_alloc_size >= 4*internal::Vector::minimum_parallel_grain_size)
+          thread_loop_partitioner.reset(new parallel::internal::TBBPartitioner());
```

Comment:
When initializing a `ReadWriteVector` with the default constructor, we will not initialize the `TBBPartitioner` variable. Thus, when we come in in this function we can still have no partitioner variable if `new_alloc_size` is less than 4 times the minimum grain size. The functions are currently not using the partitioner in that case but I think it would still be safer if either the default constructor sets the partitioner to a TBBPartitioner object or calls reinit on size 0.